test/support/utils.rb

```diff
@@ -12,7 +12,14 @@ def type(command)
     end
 
     def create_message fail_msg, test_info
-      "#{fail_msg} on #{test_info.mode} mode\n[DEBUGGER SESSION LOG]\n> #{test_info.backlog.join('> ')}#{debuggee_backlog test_info}"
+      <<~MSG.chomp
+        [DEBUGGER SESSION LOG]
```

Comment:
Maybe, a line-break is needed here? When I try your change, the output is as follows. `[DEBUGGER SESSION LOG]` is on the same line as `#<Test::Unit::AssertionFailedError:`.

```shell
Failure: test_quit_does_not_quit_debugger_process_if_not_confirmed(DEBUGGER__::QuitTest):
  #<Test::Unit::AssertionFailedError: [DEBUGGER SESSION LOG]
  > DEBUGGER: Session start (pid: 2021)
  > [1, 1] in /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210829-1993-vgyxzr.rb
  > =>   1| a=1
  > =>#0	<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210829-1993-vgyxzr.rb:1
  > q
  > n
  > (rdbg) q
Really quit? [Y/n] n

  =================================== Failure Message ===================================
  Expected to include `/Really quita\? \[Y\/n\]/` in
  (
  n
  (rdbg) q
Really quit? [Y/n] n
  )
   on LOCAL mode.>.
/Users/naotto/workspace/debug/test/support/assertions.rb:66:in `assert_block'
/Users/naotto/workspace/debug/test/support/utils.rb:69:in `rescue in debug_code'
/Users/naotto/workspace/debug/test/support/utils.rb:57:in `debug_code'
test/debug/quit_test.rb:22:in `test_quit_does_not_quit_debugger_process_if_not_confirmed'
     19:     end
     20:
     21:     def test_quit_does_not_quit_debugger_process_if_not_confirmed
  => 22:       debug_code(program) do
     23:         type 'q'
     24:         assert_line_text(/Really quita\? \[Y\/n\]/)
     25:         type 'n'
```

Reply:
👍 

I've given debugger and debuggee sessions their own block title now. So the smaller `[DEBUGGER SESSION LOG]` is now removed.

```
  ======================
  ▼  Debugger Session  ▼
  ======================

  > [1, 10] in /var/folders/yg/hnbymwxd5pn7v94_clc59y6r0000gn/T/debugger20210829-41581-1exxyh.rb
  > =>   1| class Foo
  >      2|   def bar; end
  >      3|   def self.baz; end
  >      4| end
  >      5|
  >      6| class Bar < Foo
  >      7| end
  >      8|
  >      9| f = Foo.new
  >     10| b = Bar.new
  > =>#0        <main> at /var/folders/yg/hnbymwxd5pn7v94_clc59y6r0000gn/T/debugger20210829-41581-1exxyh.rb:1
  > c
  >
  > (rdbg:remote) c
[10, 19] in /var/folders/yg/hnbymwxd5pn7v94_clc59y6r0000gn/T/debugger20210829-41581-1exxyh.rb
  >     10| b = Bar.new
  >     11|
  >     12| def f.foobar; end
  >     13| def b.foobar; end
  >     14|
  > =>  15| binding.b
  >     16|
  >     17|  Foo.baz
  >     18|  f.bar
  >     19|  f.foobar
  > =>#0        <main> at /var/folders/yg/hnbymwxd5pn7v94_clc59y6r0000gn/T/debugger20210829-41581-1exxyh.rb:15
  >
  > (rdbg:remote) trace pass Foo
Enable PassTracer for Foo (enabled)
  >
  > (rdbg:remote) trace pass fq
eval error: undefined local variable or method `fq' for main:Object
  > Did you mean?  f
  >   (rdbg)//var/folders/yg/hnbymwxd5pn7v94_clc59y6r0000gn/T/debugger20210829-41581-1exxyh.rb:1:in `<main>'
  > undefined local variable or method `fq' for main:Object
  > Did you mean?  f
  >
  > (rdbg:remote) c
DEBUGGER (trace/pass) #th:1 #depth:2 `Foo` receives .baz (#<Class:Foo>.baz) at /var/folders/yg/hnbymwxd5pn7v94_clc59y6r0000gn/T/debugger20210829-41581-1exxyh.rb:3
  > [20, 25] in /var/folders/yg/hnbymwxd5pn7v94_clc59y6r0000gn/T/debugger20210829-41581-1exxyh.rb
  >     20|
  >     21|  Bar.baz
  >     22|  b.bar
  >     23|  b.foobar
  >     24|
  > =>  25|  binding.b
  > =>#0        <main> at /var/folders/yg/hnbymwxd5pn7v94_clc59y6r0000gn/T/debugger20210829-41581-1exxyh.rb:25


  ======================
  ▲  Debugger Session  ▲
  ======================

  ===================
  ▼ Failure Message ▼
  ===================

  TIMEOUT ERROR (10 sec) on UNIX DOMAIN SOCKET mode
/Users/st0012/projects/debug/test/support/assertions.rb:66:in `assert_block'
/Users/st0012/projects/debug/test/support/utils.rb:230:in `rescue in block in run_test_scenario'
/Users/st0012/projects/debug/test/support/utils.rb:226:in `block in run_test_scenario'
/Users/st0012/projects/debug/test/support/utils.rb:180:in `spawn'
/Users/st0012/projects/debug/test/support/utils.rb:180:in `run_test_scenario'
/Users/st0012/projects/debug/test/support/utils.rb:166:in `debug_on_unix_domain_socket'
/Users/st0012/projects/debug/test/support/utils.rb:108:in `debug_code'
test/debug/trace_test.rb:317:in `test_tracer_prints_correct_method_receiving_messages'
     314:       end
     315:
     316:       def test_tracer_prints_correct_method_receiving_messages
  => 317:         debug_code(program) do
     318:           type 'c'
     319:           type 'trace pass Foo'
     320:           type 'trace pass fq'
=========================================================================================================================================================================================================================

```